js/PulsarFormComponent.js

```diff
@@ -84,33 +87,54 @@ PulsarFormComponent.prototype.initTimePickers = function () {
  */
 PulsarFormComponent.prototype.initDatePickers = function () {
     const datepickers = this.$html.find('[data-datepicker="true"]');
-    let defaultDateFormat = 'DD/MM/YYYY';
 
     datepickers.each((index, element) => {
-        let dateFormat = element.getAttribute('data-format');
+        let $element = $(element);
+        let dateFormat = 'DD/MM/YYYY';
+        let formatKey = element.getAttribute('data-format');
+        let locale = element.getAttribute('data-locale');
 
-        // Check if data-format attribute exists and lowercase it
-        // to eliminate different styles of writing issues
-        if (dateFormat !== null) {
-            dateFormat = dateFormat.toLowerCase();
+        if (typeof locale === 'string') {
+            dateFormat = moment.localeData(locale).longDateFormat('L');
         }
 
-        switch (dateFormat) {
-            case 'us':
-                defaultDateFormat = 'MM/DD/YYYY';
-                break;
-            case 'reverse':
-                defaultDateFormat = 'YYYY/MM/DD';
-                break;
-            default:
-                defaultDateFormat = 'DD/MM/YYYY';
+        if (typeof formatKey === 'string') {
+            formatKey = formatKey.toLowerCase();
+
+            switch (formatKey) {
+                case 'us':
+                    dateFormat = 'MM/DD/YYYY';
+                    break;
+                case 'reverse':
+                    dateFormat = 'YYYY/MM/DD';
+                    break;
+            }
         }
 
         // Initialize pikaday with the correct date format
-        $(element).pikaday({ format: defaultDateFormat });
+        $element.pikaday({ format: dateFormat });
 
         // Initialize placeholder attribute based on the date format
-        $(element).attr('placeholder', defaultDateFormat.toLowerCase());
+        $element.attr('placeholder', dateFormat.toLowerCase());
+
+        if (element.hasAttribute('data-iso') && element.hasAttribute('name')) {
+            let elementName = $element.attr('name');
+            let hiddenElement = $('<input>', {'type': 'hidden', 'name': elementName});
+
+            $element.removeAttr('name');
+            $element.after(hiddenElement);
+
+            $element.change(function() {
+                let parsed = moment($(this).val(), dateFormat);
+                let ISOTime = parsed.toISOString(true);
+
+                if (ISOTime !== null) {
+                    ISOTime = ISOTime.substr(0, 10);
+                }
```

Comment:
Could we use Moment's format method here instead of the substr? Taking it a step further we could have that format an option? So we'd be able to configure the value sent to the server when using the component perhaps.

Reply:
I use the `.toISOString()` method from moment before the string manipulation. The whole point is that it posts back in the ISO format - in the same way an HTML5 input does.

I can change the string manipulation for one of these though if that's more comfortable?
https://stackoverflow.com/a/39736368/1600814

Reply:
Although there are two other questions around this:
1) Does Pulsar support a `type="datetime"` with pikaday? If so, this needs to be aware of that.
2) In the `.toISOString()` I set the `keepOffset` flag to  `true` - I don't know if this is correct or not.

Reply:
On the first point, that SO answer is what I'm suggesting, particularly this `date.utc().format('HH:mm:ss');` as we can have the HTML5 format as the default, but then make it extensible by having an option to set that format string (HH:mm:ss), so when we innevitably need a non HTML5 standard date we can conifigure that per component.

On omitting the `true` argument to `toISOString` it's a tricky one... as we're stripping all the useful information out of the timestring (thanks HTML5!) we'd have to couple the understaning of the timezone between the client and server in the component itself. I'd suggest making this configurable as part of the options also, so we can decide on an implementation basis.

The pickaday question I'm not sure on, one to check for sure ⏲

Reply:
> 1. Does Pulsar support a `type="datetime"` with pikaday? If so, this needs to be aware of that.

No, I don't believe we do. There may well be datetime pickers in the wild, but not provided by Pulsar, apart from the new Symfony forms branch, but that's not Pikaday.

Reply:
In that case, it seems reasonable to me to use `date.format('YYYY-MM-DD');`. Unsure how/if the `.utc()` is strictly necessary as we're only dealing with dates I don't think we care about the timezone as we need to literally say which day not which hour.

I can't see any reasoning to make the format configurable. We should be standardising on the HTML5 format so that we are able to easily replace the text input with the date input for accessibility when the time comes.

Reply:
Good point on the `utc()` I blindly copied that for the SO you linked, I was getting mixed up there with dates and datetimes. UTC can almost certainly be omitted.

In regards to the configurable format, I would still have it as a configurable option, with maybe some accompanying best practice in the documentation. I agree that it _should_ follow the HTML5 spec.